client/securedrop_client/logic.py

```diff
@@ -842,6 +850,7 @@ def _submit_download_job(
             job.success_signal.connect(self.on_file_download_success)
             job.failure_signal.connect(self.on_file_download_failure)
 
+        job.progress = progress
```

Comment:
I was trying to figure out if there was any way of logically gating this (eg if it's a db.File then it always should have a non-null `progress` or there's an error worth logging), but that may not be true.

Reply:
Right, I think we just treat it as `ProgressProxy | None` all the way through until we actually call methods on it.